### ISSUE TITLE
Nest Link account

### ISSUE THREAD

**Issue by Himola** [hoobs-org/HOOBS#1683]
**Description**
Nest Bridge

**Expected behavoir**
I'm in the plugin configuration interface where there is a button "Link account". After I click on it a popup page appears with this info:

Login to Nest
First you will need to login to Google and get a one time access code. This will open a new dialog.
When you see the screen above, copy the code and enter it the next step.

and a login button. I click on the login button and the same popup page turns into white page and nothing happens. I've tried different browsers, same result

**Plugin name**
Nest Bridge

**Please report the issue to the plugin developer**
We like to help the community, however, some issues are out of our control. Please report this to the plugin developer and include a link here.

**Version**
4.0.71

**Did you upgrade**
Please let us know if you upgraded from a previous version.
- [ ] Yes
- [x ] No

**Pervious version**
If you upgraded, please let us know your previous version.

**Did you orginally upgrade to HOOBS 3 from HOOBS 2.1.1?**
Did you perform an upgrade from HOOBS 2.1.1? Some things are different.
- [ ] Yes
- [x ] No

**What device are you using?**
Please let us know the device you are running HOOBS on.
- [ ] HOOBS Box
- [ ] HOOBS SD Card
- [x ] Custom Install

**If this is a custom install, what's your operating system?**
Raspberry Pi OS

**Post your config**

hoobs@hoobs:~ $ cat ~/.hoobs/etc/config.json
cat: /home/hoobs/.hoobs/etc/config.json: No such file or directory
hoobs@hoobs:~ $

* Remove any passwords from the config file. This is public.

```
POST CONFIG HERE
```

**Post your log**
You can get the log from the intercafe. This is in the Log section.

```
POST LOG HERE
```

**Additional context**

**Comment by mkellsy** [comment]
Google uses popups, you might have to disable your popup blocker or add an exception.

**Comment by Himola** [comment]
I tried everything and I could not make it work. Is there any other way? 
In Homebridge there is an option to add Goggle refresh Token....it 
works... how about in Hoobs?

--

Robert PETRACHE


------ Original Message ------

**Comment by mkellsy** [comment]
Just open the terminal and run the command from the readme.
https://github.com/chrisjshull/homebridge-nest

You can manually edit the config, go to the Next tab in the Config and click the Advanced button.

**Comment by Himola** [comment]
i tried that already....in advanced tab it doesn;t save the token.

It feels like you don;t really wanna help here....because pointing me to 
some helpers online witch I already wrote is not the solution.

Thanks for your time. I'm happy I did not buy the product yet....with 
this support.

--

Robert PETRACHE


------ Original Message ------

**Comment by mkellsy** [comment]
Can you post the config you are trying to save. Replace the token with ***

**Comment by Himola** [comment]
{
     "accessories": [],
     "platforms": [
         {
             "platform": "Nest"
             "refreshToken": "xxx"
         }
     ]
}

--

Robert PETRACHE


------ Original Message ------

**Comment by mkellsy** [comment]
Your JSON is missing a comma after **"platform": "Nest**.
```
{
     "accessories": [],
     "platforms": [
         {
             "platform": "Nest",
             "refreshToken": "xxx"
         }
     ]
}
```

If you click the Advanced button inside the Nest plugin on the Config screen the JSON will be this.

```
{
    "platform": "Nest",
    "refreshToken": "xxx"
}
```

**Comment by Himola** [comment]
ok, now it was possible to save but I have this error in logs:

12/7/2021, 6:01:05 PMNest Bridge starting
12/7/2021, 6:01:08 PMNest BridgeLoaded plugin 'homebridge-nest'
12/7/2021, 6:01:08 PMNest BridgeLoading 1 platforms...
12/7/2021, 6:01:08 PMNest BridgeNestFetching Nest devices.
12/7/2021, 6:01:08 PMNest BridgeBridge is running on port 52826.
12/7/2021, 6:01:08 PMNest BridgeNestERRORNOTE: Because we couldn't 
connect to the Nest service, your Nest devices in HomeKit will not be 
responsive.
12/7/2021, 6:01:08 PMNest BridgeNestERRORUnable to authenticate with 
Google/Nest.
12/7/2021, 6:01:08 PMNest BridgeNestERRORAccess token acquisition via 
googleAuth failed (code 400).


My config is noe like this:

{
     "accessories": [],
     "platforms": [
         {
             "platform": "Nest",
             "refreshToken": "xxx"
         }
     ]
}
Token was acquired by running: node login.js in 
/var/lib/hoobs/nestbridge/node_modules/homebridge-nest/ and following 
the provided link to get Google's token.

So, what's wrong now?

Thanks,

--

Robert PETRACHE


------ Original Message ------

**Comment by mkellsy** [comment]
Have you tried using the desktop app?
https://support.hoobs.org/downloads/desktop

The Link Account button should work.

**Comment by Himola** [comment]
Well.....that turned out into another nightmare:

1. I've installed the app
2. I've unlinked the google account
3. I've tried to link it again but it told me the bridge is stopping
4. I've stopped the bridge
5. I've tried to link it again but it told me the bridge was 
starting....and it kept going like this
6. I've uninstalled the Nest plugin
7. I've removed the Nest bridge manually
8, Now I'm trying to reinstall Nest plugin, after few restarts and I get 
the message "Unable to install plugin." See it in the attached picture.

Kind regards,
--

Robert PETRACHE


------ Original Message ------
